### ISSUE TITLE
GiantBomb API doesn't find images

### ISSUE THREAD

**Issue by bruny** [maloep/romcollectionbrowser#351]
Need to add support to finding images (covers, screenshots, etc) in GiantBomb API

The payload is quite complicated to deal with, so might need some ideas. Also see #350

**Comment by maloep** [comment]
Problem is to find platform specific artwork, especially for box art. Screenshots could be ok to mix up.

If we want to be platform specific (also for other data like release date) we should try to find the correct release of a game first. We could use this as new search url to get the platform specific release instead of the game:
https://www.giantbomb.com/api/releases/?api_key=key&filter=platform:1,name:Desert%20Strike&field_list=name,id,guid

In this response there is a link to the "main" image of the release. I think this should always be the box front image.

in the next step we could grab other game data that does not need to be platform specific with the game detail url (included in the release response):
https://www.giantbomb.com/api/game/3030-1/?api_key=key

We could grab other artwork from the game response using the image tags or filter for image tags like this:
https://www.giantbomb.com/api/images/3030-1/?filter=image_tag%3AScreenshots&api_key=key

But now we won't be sure that the artwork is platform specific. We could ignore this or just leave screenshots out and only grab box art.

**Comment by maloep** [comment]
I implemented it as described above but I ran into another problem. The above approach is not very tolerant with varying game names. It will miss games when the search string does not exactly match the name in GBs database. The search method is more tolerant and will return a wider set of results. Unfortunately the search method only returns games, not releases. So, until now I did not find a satisfying solution.

- If we use the game result, we will get images that are not platform safe.
- The game result contains releases that have the information we want. But the release list in game result  just contains the link and the id of the release. No hint or name of the platform. We would need to query every single release to check if it is for the platform that we are looking for.
- Or we do it as described above and deal with missing games
